src/core/Akka/Actor/CoordinatedShutdown.cs

```diff
@@ -270,7 +270,7 @@ public class ClusterJoinUnsuccessfulReason : Reason
         /// </summary>
         internal readonly List<string> OrderedPhases;
 
-        private readonly ConcurrentBag<Func<Task<Done>>> _clrShutdownTasks = new ConcurrentBag<Func<Task<Done>>>();
```

Comment:
This is the real cause of the memory leak. `ConcurrentBag` uses `ThreadLocal` that pinned the lambdas in memory, preventing them from being GC-ed.

Reply:
yikes